### PR TITLE
Add a command to show the URL for the current file in Git

### PR DESCRIPTION
**Pull request by pgr0ss**
# What

Prints the Git Web URL for the current file at the current selection.
- Parses and constructs the remote Git URL
- Links to either the current line or a range with the current selection
- Uses `echomsg` to save the URL in the message-history

# Why

I have a common workflow where I'm investigating something in vim and then I want to send a link of what I'm looking at to someone. 

This change lets me run `:GitWebURL`, copy the URL, and then send it to someone (e.g. via slack). I can also cmd+click the URL which will open it in my browser.

I opted not to make a leader command for it since I'm not sure if it's useful to others. I also opted not to make it a plugin since it was easier to iterate in the vimrc. We can always extract it later if desired.

# Checklist

- [ ] Send RFC email to Braintree developers _if change may be impactful_. Please include a link to this PR so that disucssion about the pros and cons of the change remains linked to the proposed changes. **Recent examples include**: addition of linter and completer, no longer removing end-of-line whitespace on save, changing to fzf for file lookup.

I don't think this change is impactful enough to email everyone.

My vimscript is pretty weak, so please let me know if there are better ways to do this.

### CODE REVIEW

File: vimrc [hunk 1]
```diff
@@ -272,6 +272,15 @@ endfunction
 command! -nargs=0 GitGrepWord :call GitGrepWord()
 nnoremap <silent> <Leader>gw :GitGrepWord<CR>
 
+function! GitWebURL() range
```
Comment (leighmcculloch) [inline vimrc]: This should probably be `GitHubWebURL` given that it is GitHub specific.

Comment (pgr0ss) [reply vimrc]: Good point. I renamed it to `GitHubURL`: d500706

File: vimrc [hunk 2]
```diff
@@ -272,14 +272,14 @@ endfunction
 command! -nargs=0 GitGrepWord :call GitGrepWord()
 nnoremap <silent> <Leader>gw :GitGrepWord<CR>
 
-function! GitWebURL() range
+function! GitHubURL() range
   let repo = systemlist("git config --get remote.origin.url | sed 's/\.git$//' | sed 's_^git@\\(.*\\):_https://\\1/_'")[0]
```
Comment (parrish) [inline vimrc]: On my local repo, I have the braintreeps repo as an upstream remote:

```
[remote "origin"]
  url = git@github.com:parrish/vim_dotfiles.git
  fetch = +refs/heads/*:refs/remotes/origin/*
[remote "upstream"]
  url = https://github.com/braintreeps/vim_dotfiles.git
  fetch = +refs/heads/*:refs/remotes/upstream/*
[branch "git_web_url"]
  remote = upstream
  merge = refs/heads/git_web_url
```

So, I took a stab at getting it use the current remote:

```
let repo = systemlist("git remote get-url `git config branch.$(git name-rev --name-only HEAD).remote` | sed 's/\.git$//' | sed 's_^git@\\(.*\\):_https://\\1/_'")[0]
```

There's probably a more elegant way to do that though

Comment (pgr0ss) [reply vimrc]: Good call. Some of our machines seem to have an older version of git without `git remote get-url` so I solved it a different way: 33d63d1

**Pull request merged**